x-pack/plugins/enterprise_search/public/applications/shared/kibana_breadcrumbs/generate_breadcrumbs.test.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { appSearchBreadcrumbs, enterpriseSearchBreadcrumbs } from '../kibana_breadcrumbs';
+
+jest.mock('../react_router_helpers', () => ({
+  letBrowserHandleEvent: () => false,
+}));
+
+describe('appSearchBreadcrumbs', () => {
+  const historyMock = {
+    createHref: jest.fn().mockImplementation(path => path.pathname),
+    push: jest.fn(),
+  };
+
+  const breadCrumbs = [
+    {
+      text: 'Page 1',
+      path: '/page1',
+    },
+    {
+      text: 'Page 2',
+      path: '/page2',
+    },
+  ];
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const subject = () => appSearchBreadcrumbs(historyMock)(breadCrumbs);
+
+  it('Builds a chain of breadcrumbs with Enterprise Search and App Search at the root', () => {
```

Comment:
The tests in this file feel a bit like testing the presentation, like a thinly veiled snapshot test. I question if this test would have a long shelf life. It also doesn't test any edge cases. Maybe a higher-level jest integration test, one that actually tests `letBrowserHandleEvent`, but still one that mocks Kibana, and interacts with the page, would provide more value. 🤷 
Was this test helpful in TDD'ing `appSearchBreadcrumbs`?

Reply:
This test will do 2 things:

- Documents what exactly the `appSearchBreadcrumbs` method does. It is not apparent by looking at the code, and it wasn't until I wrote this test that I understood what exactly it does. So it will serve as a helpful reference if anyone needs to go change this code at some point.
- Additionally, if someone ever needs to go modify this code or refactor it, these tests will help them to be confident that their changes are working as expected.

I personally like this test, it gives me confidence about what this unit does and that it is working as expected.

As to your suggestion about a higher level jest test, what would that look like? Would that just be this test with `letBrowserHandleEvent` un-mocked? What else would need to change?

Reply:
> Documents what exactly the appSearchBreadcrumbs method does

👍  "Tests as documentation for what the code does" -- that's another value we should pin somewhere in our eventual Testing Principles Guide.

I went back and looked at the corresponding `appSearchBreadcrumbs` implementation. Now, seeing how it is written, I also like this test. I see that the test passes additional breadcrumb configuration into the method, and this test makes it clear the order in which operations are intended to be applied.

I (mistakenly) originally thought this was testing only the hard-coded configuration of the breadcrumbs (e.g. `/` comes first, `/app_search` comes next), which I assumed would be brittle. The test serves a deeper purpose than that. Comment redacted :)